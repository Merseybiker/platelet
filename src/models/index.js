// @ts-check
import { initSchema } from '@aws-amplify/datastore';
import { schema } from './schema';

const Role = {
  "COORDINATOR": "COORDINATOR",
  "RIDER": "RIDER",
  "ADMIN": "ADMIN"
};

const Patch = {
  "NORTH": "NORTH",
  "WEST": "WEST",
  "EAST": "EAST",
  "SOUTH": "SOUTH",
  "RELIEF": "RELIEF",
  "AIR_AMBULANCE": "AIR_AMBULANCE"
};

const Priority = {
  "HIGH": "HIGH",
  "MEDIUM": "MEDIUM",
  "LOW": "LOW"
};

const DeliverableUnit = {
  "LITRE": "LITRE",
  "MILLILITRES": "MILLILITRES",
  "GRAMS": "GRAMS",
  "COUNT": "COUNT",
  "BOX": "BOX"
};

const TaskStatus = {
  "NEW": "NEW",
  "ACTIVE": "ACTIVE",
  "PICKED_UP": "PICKED_UP",
  "DROPPED_OFF": "DROPPED_OFF",
  "CANCELLED": "CANCELLED",
  "REJECTED": "REJECTED"
};

const { User, AddressAndContactDetails, Vehicle, Comment, Group, UserTasks, Task, CoordinatorTasks, Deliverable, DeliverableType, Location } = initSchema(schema);

export {
  User,
  AddressAndContactDetails,
  Vehicle,
  Comment,
  Group,
  UserTasks,
  Task,
  CoordinatorTasks,
  Deliverable,
  DeliverableType,
  Location,
  Role,
  Patch,
  Priority,
  DeliverableUnit,
  TaskStatus
};