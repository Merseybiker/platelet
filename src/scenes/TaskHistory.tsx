import React from "react";
import { Skeleton, Stack } from "@mui/material";
import { useInView } from "react-intersection-observer";
import useGetTasksGraphql from "../hooks/useGetTasksGraphql";
import _ from "lodash";

const TaskHistory: React.FC = () => {
    const { state, getNext, finished } = useGetTasksGraphql();
    const { ref, inView } = useInView({
        threshold: 0,
    });

    React.useEffect(() => {
        if (inView && !finished) {
            getNext();
        }
    }, [inView, getNext, state, finished]);
    return (
        <>
            <Stack>
                {state.map((task) => (
                    <div>{task.id}</div>
                ))}
                <div ref={ref} />
            </Stack>
            {!finished && (
                <Stack>
                    {_.range(0, 10).map((i) => (
                        <Skeleton key={i} />
                    ))}
                </Stack>
            )}
        </>
    );
};

export default TaskHistory;
