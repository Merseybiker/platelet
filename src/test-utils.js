import React from "react";
import rootReducer from "./redux/Reducers";
import createSagaMiddleware from "redux-saga";
import {
    ThemeProvider,
    StyledEngineProvider,
    createTheme,
} from "@mui/material/styles";
import { CssBaseline } from "@mui/material";
import { SnackbarProvider } from "notistack";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import ReactNotification from "react-notifications-component";
import "react-notifications-component/dist/theme.css";
import AdapterDateFns from "@mui/lab/AdapterDateFns";
import LocalizationProvider from "@mui/lab/LocalizationProvider";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import App from "./App";

const taskStatus = {
    NEW: "rgba(252, 231, 121, 1)",
    ACTIVE: "cornflowerblue",
    PICKED_UP: "orange",
    DROPPED_OFF: "lightgreen",
    CANCELLED: "blue",
    REJECTED: "grey",
};
const theme = createTheme({
    palette: {
        mode: "light",
        background: {
            default: "rgb(235, 235, 235)",
        },
        taskStatus,
    },
});

function getRandomInt(min, max) {
    min = Math.ceil(min);
    max = Math.floor(max);
    return Math.floor(Math.random() * (max - min) + min); //The maximum is exclusive and the minimum is inclusive
}

export function generateTimes(previous = null, hours = 2) {
    let date;
    if (previous) date = new Date(previous);
    else date = new Date();
    if (!previous) date.setHours(date.getHours() - hours);
    date.setMinutes(date.getMinutes() + getRandomInt(20, 30));
    const timeOfCall = date.toISOString();
    date.setMinutes(date.getMinutes() + getRandomInt(20, 30));
    const timePickedUp = date.toISOString();
    date.setMinutes(date.getMinutes() + getRandomInt(20, 30));
    const timeDroppedOff = date.toISOString();

    return { timeDroppedOff, timePickedUp, timeOfCall };
}

const sagaOptions = {
    onErraor: (action, error) => {
        console.log("An uncaught exception has occurred in redux-saga:");
        console.log(action);
        if (error) {
            console.log(error.message);
        }
        throw error;
    },
};

const sagaMiddleWare = createSagaMiddleware(sagaOptions);

function render(
    ui,
    {
        preloadedState,
        store = configureStore({
            reducer: rootReducer,
            preloadedState,
            middleware: [sagaMiddleWare],
        }),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return (
            <Provider store={store}>
                <BrowserRouter>
                    <LocalizationProvider dateAdapter={AdapterDateFns}>
                        <ReactNotification />
                        <App>{children}</App>
                    </LocalizationProvider>
                </BrowserRouter>
            </Provider>
        );
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };

const AllTheProviders = ({ children }) => {
    return (
        <Provider store={null}>
            <BrowserRouter>
                <LocalizationProvider dateAdapter={AdapterDateFns}>
                    <ReactNotification />
                    <App>{children}</App>
                </LocalizationProvider>
            </BrowserRouter>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: AllTheProviders, ...options });
