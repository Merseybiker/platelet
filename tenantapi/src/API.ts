/* tslint:disable */
/* eslint-disable */
//  This file was automatically generated and should not be edited.

export type CreateTenantInput = {
  id?: string | null,
  name: string,
  config: string,
};

export type ModelTenantConditionInput = {
  name?: ModelStringInput | null,
  config?: ModelStringInput | null,
  and?: Array< ModelTenantConditionInput | null > | null,
  or?: Array< ModelTenantConditionInput | null > | null,
  not?: ModelTenantConditionInput | null,
};

export type ModelStringInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export enum ModelAttributeTypes {
  binary = "binary",
  binarySet = "binarySet",
  bool = "bool",
  list = "list",
  map = "map",
  number = "number",
  numberSet = "numberSet",
  string = "string",
  stringSet = "stringSet",
  _null = "_null",
}


export type ModelSizeInput = {
  ne?: number | null,
  eq?: number | null,
  le?: number | null,
  lt?: number | null,
  ge?: number | null,
  gt?: number | null,
  between?: Array< number | null > | null,
};

export type Tenant = {
  __typename: "Tenant",
  id: string,
  name: string,
  config: string,
  createdAt: string,
  updatedAt: string,
};

export type UpdateTenantInput = {
  id: string,
  name?: string | null,
  config?: string | null,
};

export type DeleteTenantInput = {
  id: string,
};

export type ModelTenantFilterInput = {
  id?: ModelIDInput | null,
  name?: ModelStringInput | null,
  config?: ModelStringInput | null,
  and?: Array< ModelTenantFilterInput | null > | null,
  or?: Array< ModelTenantFilterInput | null > | null,
  not?: ModelTenantFilterInput | null,
};

export type ModelIDInput = {
  ne?: string | null,
  eq?: string | null,
  le?: string | null,
  lt?: string | null,
  ge?: string | null,
  gt?: string | null,
  contains?: string | null,
  notContains?: string | null,
  between?: Array< string | null > | null,
  beginsWith?: string | null,
  attributeExists?: boolean | null,
  attributeType?: ModelAttributeTypes | null,
  size?: ModelSizeInput | null,
};

export type ModelTenantConnection = {
  __typename: "ModelTenantConnection",
  items:  Array<Tenant | null >,
  nextToken?: string | null,
};

export type CreateTenantMutationVariables = {
  input: CreateTenantInput,
  condition?: ModelTenantConditionInput | null,
};

export type CreateTenantMutation = {
  createTenant?:  {
    __typename: "Tenant",
    id: string,
    name: string,
    config: string,
    createdAt: string,
    updatedAt: string,
  } | null,
};

export type UpdateTenantMutationVariables = {
  input: UpdateTenantInput,
  condition?: ModelTenantConditionInput | null,
};

export type UpdateTenantMutation = {
  updateTenant?:  {
    __typename: "Tenant",
    id: string,
    name: string,
    config: string,
    createdAt: string,
    updatedAt: string,
  } | null,
};

export type DeleteTenantMutationVariables = {
  input: DeleteTenantInput,
  condition?: ModelTenantConditionInput | null,
};

export type DeleteTenantMutation = {
  deleteTenant?:  {
    __typename: "Tenant",
    id: string,
    name: string,
    config: string,
    createdAt: string,
    updatedAt: string,
  } | null,
};

export type GetTenantQueryVariables = {
  id: string,
};

export type GetTenantQuery = {
  getTenant?:  {
    __typename: "Tenant",
    id: string,
    name: string,
    config: string,
    createdAt: string,
    updatedAt: string,
  } | null,
};

export type ListTenantsQueryVariables = {
  filter?: ModelTenantFilterInput | null,
  limit?: number | null,
  nextToken?: string | null,
};

export type ListTenantsQuery = {
  listTenants?:  {
    __typename: "ModelTenantConnection",
    items:  Array< {
      __typename: "Tenant",
      id: string,
      name: string,
      config: string,
      createdAt: string,
      updatedAt: string,
    } | null >,
    nextToken?: string | null,
  } | null,
};

export type OnCreateTenantSubscription = {
  onCreateTenant?:  {
    __typename: "Tenant",
    id: string,
    name: string,
    config: string,
    createdAt: string,
    updatedAt: string,
  } | null,
};

export type OnUpdateTenantSubscription = {
  onUpdateTenant?:  {
    __typename: "Tenant",
    id: string,
    name: string,
    config: string,
    createdAt: string,
    updatedAt: string,
  } | null,
};

export type OnDeleteTenantSubscription = {
  onDeleteTenant?:  {
    __typename: "Tenant",
    id: string,
    name: string,
    config: string,
    createdAt: string,
    updatedAt: string,
  } | null,
};
