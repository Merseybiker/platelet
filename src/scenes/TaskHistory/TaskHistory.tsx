import React from "react";
import { Box, Skeleton, Stack, Typography } from "@mui/material";
import { useInView } from "react-intersection-observer";
import useGetTasksGraphql from "../../hooks/useGetTasksGraphql";
import _ from "lodash";
import TaskHistoryCard from "./components/TaskHistoryCard";
import TaskHistoryControls from "./components/TaskHistoryControls";
import { ModelSortDirection } from "../../API";
import DateStampDivider from "../Dashboard/components/TimeStampDivider";
import TaskHistoryTaskDialog from "./components/TaskHistoryTaskDialog";

const limit = 20;

const TaskHistory: React.FC = () => {
    let displayDate = false;
    let lastTime = new Date();
    const [sortDirection, setSortDirection] =
        React.useState<ModelSortDirection>(ModelSortDirection.DESC);
    const [dateRange, setDateRange] = React.useState<{
        startDate: Date | null;
        endDate: Date | null;
    }>({
        startDate: new Date("2000-01-01"),
        endDate: new Date(),
    });
    const [openTaskId, setOpenTaskId] = React.useState<string | null>(null);
    const { state, getNext, isFinished, isFetching, error } =
        useGetTasksGraphql(
            limit,
            sortDirection,
            dateRange.startDate,
            dateRange.endDate
        );
    const { ref, inView } = useInView({
        threshold: 0,
    });

    const handleSetDateRange = (
        startDate: Date | null,
        endDate: Date | null
    ) => {
        setDateRange({ startDate, endDate });
    };

    React.useEffect(() => {
        if (inView && !isFinished) {
            getNext();
        }
    }, [inView, getNext, state, isFinished]);

    const skeletonRange = state.length ? 1 : 20;

    return (
        <>
            <Stack spacing={1}>
                <TaskHistoryControls
                    sortDirection={sortDirection}
                    setSortDirection={(v) => setSortDirection(v)}
                    setDateRange={handleSetDateRange}
                    isFetching={isFetching}
                />
                {state.map((task) => {
                    displayDate = false;
                    let component = (
                        <Box
                            onClick={() => setOpenTaskId(task.id)}
                            key={task.id}
                        >
                            <TaskHistoryCard task={task} />
                        </Box>
                    );
                    if (task.dateCreated) {
                        const timeComparison = new Date(task.dateCreated);
                        if (
                            timeComparison.getDate() <=
                            lastTime.getDate() - 1
                        ) {
                            lastTime = timeComparison;
                            displayDate = true;
                        }
                        if (displayDate) {
                            component = (
                                <Box
                                    onClick={() => setOpenTaskId(task.id)}
                                    key={task.id}
                                >
                                    <Box sx={{ maxWidth: 800 }}>
                                        <DateStampDivider
                                            date={lastTime.toISOString()}
                                        />
                                    </Box>
                                    <TaskHistoryCard task={task} />
                                </Box>
                            );
                        }
                    }
                    return component;
                })}
                {error && <Typography>Sorry, something went wrong.</Typography>}
                <div ref={ref} />
                {!isFinished && (
                    <Stack spacing={1} data-testid="task-history-skeleton">
                        {_.range(0, skeletonRange).map((i) => (
                            <Box key={i} sx={{ maxWidth: 1200 }}>
                                <Skeleton
                                    variant="rectangular"
                                    height={140}
                                    sx={{
                                        borderRadius: 4,
                                    }}
                                />
                            </Box>
                        ))}
                    </Stack>
                )}
            </Stack>
            {openTaskId && (
                <TaskHistoryTaskDialog
                    taskId={openTaskId}
                    onClose={() => setOpenTaskId(null)}
                />
            )}
        </>
    );
};

export default TaskHistory;
