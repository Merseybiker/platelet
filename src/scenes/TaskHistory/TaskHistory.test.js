import TaskHistory from "./TaskHistory";
import * as models from "../../models";
import { mockAllIsIntersecting } from "react-intersection-observer/test-utils";
import { render } from "../../test-utils";
import { API, DataStore } from "aws-amplify";
import { screen, waitFor } from "@testing-library/react";
import * as APITypes from "../../API";

const tenantId = "testTenantId";

let preloadedState = {
    tenantId,
};

describe("TaskHistory", () => {
    let mockTask = {};
    const isoDate = "2021-11-29T23:24:58.987Z";
    beforeAll(() => {
        jest.useFakeTimers("modern");
        jest.setSystemTime(new Date(isoDate));
    });

    afterAll(() => {
        jest.useRealTimers();
    });

    beforeEach(async () => {
        jest.restoreAllMocks();
    });

    beforeAll(async () => {
        const mockWhoami = await DataStore.save(
            new models.User({
                tenantId,
                displayName: "someDisplayName",
            })
        );
        preloadedState = { ...preloadedState, whoami: { user: mockWhoami } };
        mockTask = {
            id: "someId",
            timeOfCall: "2021-05-01T00:00:00.000Z",
            dateCreated: "2021-05-01",
            riderResponsibility: "someRiderResponsibility",
            priority: APITypes.Priority.HIGH,
            status: APITypes.TaskStatus.COMPLETED,
            createdAt: "2021-05-01T00:00:00.000Z",
            assignees: {
                items: [
                    {
                        assignee: {
                            displayName: "someDisplayName",
                            profilePicture: {
                                key: "someKey",
                            },
                        },
                    },
                ],
            },
            pickUpLocation: {
                name: "pickup name",
                ward: "pickup ward",
                line1: "pickup line1",
                town: "pickup town",
                postcode: "pickup postcode",
            },
            dropOffLocation: {
                ward: "dropoff ward",
                line1: "dropoff line1",
                name: "dropoff name",
                town: "dropoff town",
                postcode: "dropoff postcode",
            },
            deliverables: {
                items: [
                    {
                        count: 1,
                        deliverableType: {
                            label: "someDeliverable",
                            icon: APITypes.DeliverableTypeIcon.BUG,
                        },
                    },
                ],
            },
            comments: {
                items: [
                    {
                        id: "someId",
                        visibility: APITypes.CommentVisibility.EVERYONE,
                        author: { id: mockWhoami.id },
                    },
                    {
                        id: "someId",
                        visibility: APITypes.CommentVisibility.EVERYONE,
                        author: { id: "another" },
                    },
                    {
                        id: "someId",
                        visibility: APITypes.CommentVisibility.EVERYONE,
                        author: { id: mockWhoami.id },
                    },
                    {
                        id: "someId",
                        visibility: APITypes.CommentVisibility.ME,
                        author: { id: "another" },
                    },
                ],
            },
            _version: 1,
            _deleted: null,
            _lastChangedAt: 1620000000000,
        };
    });

    it("displays task details", async () => {
        const graphqlSpy = jest.spyOn(API, "graphql").mockResolvedValueOnce({
            data: {
                getTasksByTenantId: {
                    items: [mockTask],
                    nextToken: null,
                    startedAt: 1620000000000,
                },
            },
        });
        render(<TaskHistory />, { preloadedState });
        mockAllIsIntersecting(true);
        await waitFor(() => {
            expect(screen.queryByTestId("task-history-skeleton")).toBeNull();
        });
        expect(graphqlSpy).toMatchSnapshot();
        expect(screen.getByText("HIGH")).toBeInTheDocument();
        expect(screen.getByText("COMPLETED")).toBeInTheDocument();
        expect(screen.getByText("someDisplayName")).toBeInTheDocument();
        expect(screen.getByText("someDeliverable x 1")).toBeInTheDocument();
        expect(screen.getByText("someRiderResponsibility")).toBeInTheDocument();
        expect(screen.getByText("pickup name")).toBeInTheDocument();
        expect(
            screen.getByText(
                "pickup ward, pickup line1, pickup town, pickup postcode"
            )
        ).toBeInTheDocument();
        expect(screen.getByText("dropoff name")).toBeInTheDocument();
        expect(
            screen.getByText(
                "dropoff ward, dropoff line1, dropoff town, dropoff postcode"
            )
        ).toBeInTheDocument();
        expect(screen.getByText("3")).toBeInTheDocument();
    });
    it("paginates results", async () => {
        const secondTask = {
            ...mockTask,
            status: APITypes.TaskStatus.PICKED_UP,
        };
        const graphqlSpy = jest
            .spyOn(API, "graphql")
            .mockResolvedValueOnce({
                data: {
                    getTasksByTenantId: {
                        items: [mockTask],
                        nextToken: "someNextToken",
                        startedAt: 1620000000000,
                    },
                },
            })
            .mockResolvedValueOnce({
                data: {
                    getTasksByTenantId: {
                        items: [secondTask],
                        nextToken: null,
                        startedAt: 1620000000001,
                    },
                },
            });
        render(<TaskHistory />, { preloadedState });
        expect(await screen.findByText("COMPLETED")).toBeInTheDocument();
        mockAllIsIntersecting(true);
        await waitFor(() => {
            expect(screen.queryByTestId("task-history-skeleton")).toBeNull();
        });
        expect(screen.getByText("PICKED UP")).toBeInTheDocument();
        expect(graphqlSpy).toMatchSnapshot();
    });
});
