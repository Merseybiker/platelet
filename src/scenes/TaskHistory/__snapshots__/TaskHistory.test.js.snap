// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskHistory displays task details 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "query": "
    query GetTasksByTenantId(
        $filter: ModelTaskFilterInput
        $limit: Int
        $nextToken: String
        $tenantId: ID!
        $sortDirection: ModelSortDirection
        $startDate: String
        $endDate: String
    ) {
        getTasksByTenantId(
            filter: $filter
            limit: $limit
            nextToken: $nextToken
            tenantId: $tenantId
            sortDirection: $sortDirection
            dateCreated: { between: [$startDate, $endDate] }
        ) {
            items {
                id
                timeOfCall
                dateCreated
                riderResponsibility
                priority
                status
                createdAt
                assignees {
                    items {
                        assignee {
                            displayName
                            profilePicture {
                                key
                            }
                        }
                    }
                }
                pickUpLocation {
                    name
                    ward
                    line1
                    town
                    postcode
                }
                dropOffLocation {
                    ward
                    line1
                    name
                    town
                    postcode
                }
                deliverables {
                    items {
                        count
                        deliverableType {
                            label
                            icon
                        }
                    }
                }
                comments {
                    items {
                        id
                        visibility
                        author {
                            id
                        }
                    }
                }
                _version
                _deleted
                _lastChangedAt
            }
            nextToken
            startedAt
        }
    }
",
        "variables": Object {
          "endDate": "2021-11-29T23:24:58.987Z",
          "limit": 20,
          "sortDirection": "DESC",
          "startDate": "2000-01-01T00:00:00.000Z",
          "tenantId": "testTenantId",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        Symbol(async_id_symbol): 37441,
        Symbol(trigger_async_id_symbol): 37425,
        Symbol(destroyed): Object {
          "destroyed": false,
        },
      },
    },
  ],
}
`;

exports[`TaskHistory paginates results 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "query": "
    query GetTasksByTenantId(
        $filter: ModelTaskFilterInput
        $limit: Int
        $nextToken: String
        $tenantId: ID!
        $sortDirection: ModelSortDirection
        $startDate: String
        $endDate: String
    ) {
        getTasksByTenantId(
            filter: $filter
            limit: $limit
            nextToken: $nextToken
            tenantId: $tenantId
            sortDirection: $sortDirection
            dateCreated: { between: [$startDate, $endDate] }
        ) {
            items {
                id
                timeOfCall
                dateCreated
                riderResponsibility
                priority
                status
                createdAt
                assignees {
                    items {
                        assignee {
                            displayName
                            profilePicture {
                                key
                            }
                        }
                    }
                }
                pickUpLocation {
                    name
                    ward
                    line1
                    town
                    postcode
                }
                dropOffLocation {
                    ward
                    line1
                    name
                    town
                    postcode
                }
                deliverables {
                    items {
                        count
                        deliverableType {
                            label
                            icon
                        }
                    }
                }
                comments {
                    items {
                        id
                        visibility
                        author {
                            id
                        }
                    }
                }
                _version
                _deleted
                _lastChangedAt
            }
            nextToken
            startedAt
        }
    }
",
        "variables": Object {
          "endDate": "2021-11-29T23:24:59.137Z",
          "limit": 20,
          "sortDirection": "DESC",
          "startDate": "2000-01-01T00:00:00.000Z",
          "tenantId": "testTenantId",
        },
      },
    ],
    Array [
      Object {
        "query": "
    query GetTasksByTenantId(
        $filter: ModelTaskFilterInput
        $limit: Int
        $nextToken: String
        $tenantId: ID!
        $sortDirection: ModelSortDirection
        $startDate: String
        $endDate: String
    ) {
        getTasksByTenantId(
            filter: $filter
            limit: $limit
            nextToken: $nextToken
            tenantId: $tenantId
            sortDirection: $sortDirection
            dateCreated: { between: [$startDate, $endDate] }
        ) {
            items {
                id
                timeOfCall
                dateCreated
                riderResponsibility
                priority
                status
                createdAt
                assignees {
                    items {
                        assignee {
                            displayName
                            profilePicture {
                                key
                            }
                        }
                    }
                }
                pickUpLocation {
                    name
                    ward
                    line1
                    town
                    postcode
                }
                dropOffLocation {
                    ward
                    line1
                    name
                    town
                    postcode
                }
                deliverables {
                    items {
                        count
                        deliverableType {
                            label
                            icon
                        }
                    }
                }
                comments {
                    items {
                        id
                        visibility
                        author {
                            id
                        }
                    }
                }
                _version
                _deleted
                _lastChangedAt
            }
            nextToken
            startedAt
        }
    }
",
        "variables": Object {
          "endDate": "2021-11-29",
          "limit": 20,
          "nextToken": "someNextToken",
          "sortDirection": "DESC",
          "startDate": "2000-01-01",
          "tenantId": "testTenantId",
        },
      },
    ],
    Array [
      Object {
        "query": "
    query GetTasksByTenantId(
        $filter: ModelTaskFilterInput
        $limit: Int
        $nextToken: String
        $tenantId: ID!
        $sortDirection: ModelSortDirection
        $startDate: String
        $endDate: String
    ) {
        getTasksByTenantId(
            filter: $filter
            limit: $limit
            nextToken: $nextToken
            tenantId: $tenantId
            sortDirection: $sortDirection
            dateCreated: { between: [$startDate, $endDate] }
        ) {
            items {
                id
                timeOfCall
                dateCreated
                riderResponsibility
                priority
                status
                createdAt
                assignees {
                    items {
                        assignee {
                            displayName
                            profilePicture {
                                key
                            }
                        }
                    }
                }
                pickUpLocation {
                    name
                    ward
                    line1
                    town
                    postcode
                }
                dropOffLocation {
                    ward
                    line1
                    name
                    town
                    postcode
                }
                deliverables {
                    items {
                        count
                        deliverableType {
                            label
                            icon
                        }
                    }
                }
                comments {
                    items {
                        id
                        visibility
                        author {
                            id
                        }
                    }
                }
                _version
                _deleted
                _lastChangedAt
            }
            nextToken
            startedAt
        }
    }
",
        "variables": Object {
          "endDate": "2021-11-29",
          "limit": 20,
          "nextToken": "someNextToken",
          "sortDirection": "DESC",
          "startDate": "2000-01-01",
          "tenantId": "testTenantId",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {
        Symbol(async_id_symbol): 37738,
        Symbol(trigger_async_id_symbol): 37722,
        Symbol(destroyed): Object {
          "destroyed": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        Symbol(async_id_symbol): 37801,
        Symbol(trigger_async_id_symbol): 37755,
        Symbol(destroyed): Object {
          "destroyed": false,
        },
      },
    },
    Object {
      "type": "return",
      "value": Promise {
        Symbol(async_id_symbol): 37829,
        Symbol(trigger_async_id_symbol): 37803,
        Symbol(destroyed): Object {
          "destroyed": false,
        },
      },
    },
  ],
}
`;
